docs/Users/Tutorials/Testing_Bears.rst

```diff
@@ -0,0 +1,196 @@
+How to use LocalBearTestHelper to test your bears
+=================================================
+
+coala has an awesome testing framework so that you can write tests for bears
+with ease.
+
+What is LocalBearTestHelper
+---------------------------
+
+You can use the following to test your bears:
+
+- LocalBearTestHelper class
+- verify_local_bears
+
+.. note::
+
+    ``execute_bears`` is used by LocalBearTestHelper class.
+
+Understanding the LocalBearTestHelper class
+-------------------------------------------
+
+LocalBearTestHelper class is a helper class for simplifying local bear testing.
+
+This class has two methods:
+
+- check_validity()
+- check_results()
+
+``check_validity`` asserts if you're bear yields any results for a particular
+check. ``check_results`` asserts if you're bear results matches with the actual
+results on execution on CLI.
+
+.. note::
+
+    The above two methods use ``execute bear`` function.LocalBearTestHelper
+    class inherites its other methods from ``unittest.Testcase``.
+
+Understanding through examples
+------------------------------
+
+**EXAMPLE 1** using ``verify_bear``
+
+*PyDocStyleBearTest*
+
+.. code::
+
+ from bears.python.PyDocStyleBear import PyDocStyleBear
+ from tests.LocalBearTestHelper import verify_local_bear
+
+
+ good_file = '''
+ """example module level docstring."""
+
+
+ def hello():
+     """Print hello world."""
+     print("hello world")
+
+ '''.splitlines(keepends=True)
+
+
+ bad_file = '''
+ """
+ example module level docstring
+ """
+
+ def hello():
+     print("hello world")
+
+ '''.splitlines(keepends=True)
+
+ PyDocStyleBearTest = verify_local_bear(PyDocStyleBear,
+                                        valid_files=(good_file,),
+                                        invalid_files=(bad_file,),
+                                        tempfile_kwargs={"suffix": ".py"})
+
+``good_file`` is a file which follows all the checks that you're bear is
```

Comment:
"follow all the checks [...]", don't like this kind of phrasing... Maybe "what your bear considers as good/as non-style-violating / ... "?